src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,21 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && !isset($file->data[$key]) && $member_name !== null) {
+                $traitAnnotations = array();
+
+                if (isset($file) && isset($file->traitMethodOverrides[$class_name][$member_name])) {
+                    list($traitName, $originalMemberName) = $file->traitMethodOverrides[$class_name][$member_name];
+                    $traitAnnotations = $this->getAnnotations($traitName, $member_type, $originalMemberName);
+                } else {
+                    foreach ($reflection->getTraitNames() as $traitName) {
+                        $traitAnnotations += $this->getAnnotations($traitName, $member_type, $member_name);
```

Comment:
It's bad, that `getAnnotations` method itself doesn't return something special, when attempt to get annotations is made from non existing property/method. The public `getPropertyAnnotations` and `getMethodAnnotations` do that currently.

I don't want this code to turn out into something ugly, like `if ( is property ) do this else if ( is method ) do this` or `try getAnnotations catch ( method/property missing ) then do something special` either.

The optimal way would be for `getAnnotations` method to return `null` or `false` for cases, when `!isset($file->data[$key])` so that we can clearly distinguish 2 cases:

1. no annotations were found on given class/trait member (property/method)
2. given class/trait member (property/method) doesn't exist

Main point is to escape the `method_exists` and `property_exists` checks all over the `getAnnotations` method.